### PR TITLE
feat: add zhipu llm support and fix some already know bugs

### PR DESCRIPTION
**Pull request by ruanrongman**
This is closed to #22 , add zhipu llm support and fix some already know bugs

## Summary by CodeRabbit

- **New Features**
	- Integrated ZhiPu language model for enhanced interaction capabilities.
	- Added new system prompt presets for ZhiPu in multiple languages.

- **Enhancements**
	- Expanded language model configuration options.
	- Improved clarity in tool descriptions with added input details.

- **Bug Fixes**
	- Resolved an issue where a specific network error was not properly handled.

### CODE REVIEW

File: promptulate/preset_roles/prompt.py [hunk 1]
```diff
@@ -41,6 +41,19 @@
 
 """  # noqa: E501
 
+# 250+ token
+PRESET_SYSTEM_PROMPT_ZhiPU = """
```
Comment (Undertone0809) [inline promptulate/preset_roles/prompt.py:45]: Can we remove it？It's repeat and useless. One system prompt preset is enough.

File: promptulate/llms/zhipu/platform.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    MessageSet,
+    SystemMessage,
+    UserMessage,
+)
+from promptulate.utils import logger
+
+T = TypeVar("T", bound=BaseModel)
+
+
+class ZhiPuStreamIterator:
```
Comment (Undertone0809) [inline promptulate/llms/zhipu/platform.py:28]: We aim to develop a versatile Pne StreamIterator, designed to encapsulate the streaming responses from various models. Unlike the `ZhiPuStreamIterator`, which is tailored for ZhiPu-type data, or those designed for OpenAI-type data, the unique StreamIterator will be agnostic to the data source. Like `BaseMessage` type, ensuring compatibility and flexibility across different model outputs.

This revised text conveys the intention to create a flexible iterator capable of handling a standardized message format, making it adaptable to multiple sources of streamed data.

File: promptulate/llms/zhipu/platform.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        )
+        # return stream
+        if stream:
+            return ZhiPuStreamIterator(
+                response_stream=response.iter_lines(decode_unicode=True),
+                return_raw_response=return_raw_response,
+            )
+        else:
+            if response.status_code == 200:
+                # todo enable stream mode
```
Comment (Undertone0809) [inline promptulate/llms/zhipu/platform.py:229]: Remove Useless Code Comments

File: pyproject.toml [hunk 4]
```diff
@@ -35,6 +35,7 @@ python-dotenv = "^1.0.0"
 questionary = "^2.0.1"
 requests = "^2.31.0"
 jinja2 = "^3.1.3"
+pyjwt = "^2.8.0"
```
Comment (Undertone0809) [inline pyproject.toml:38]: This dependency should add in ​ [​tool​.​poetry​.​group​.​test_integration​.​dependencies​]. Run make lock after add it.

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

**Comment by Undertone0809** [comment]
It will get better if we add ZhiPu usage docs.

File: promptulate/preset_roles/prompt.py (hunk 1 above)
Comment (ruanrongman) [reply promptulate/preset_roles/prompt.py:45]: ok

File: promptulate/llms/zhipu/platform.py (hunk 3 above)
Comment (ruanrongman) [reply promptulate/llms/zhipu/platform.py:229]: ok

File: promptulate/schema.py [hunk 5]
```diff
@@ -28,6 +28,91 @@ def type(self) -> str:
...
+    Attributes:
+        response_stream: The stream of responses from the LLM model.
+        return_raw_response: A boolean indicating whether to return the raw response
+        or not.
+    """
+
+    def __init__(
+        self,
+        response_stream,
+        parse_content: callable([[Any], [str, str]]),
```
Comment (Undertone0809) [inline promptulate/schema.py:44]: You should add relevant docs to show how to pass parse_content callback.

Comment (ruanrongman) [reply promptulate/schema.py:45]: ok

**Review by Undertone0809 — APPROVED** [review]

**Pull request merged**